### ISSUE TITLE
Replace Prism with starry-night

### ISSUE THREAD

**Issue by kidonng** [denoland/deno-gfm#56]
[starry-night](https://github.com/wooorm/starry-night) is a syntax highlighter with feature parity to GitHub, which makes it an interesting alternative to currently used Prism. And styling is free if #55 is accepted.

**Comment by kidonng** [comment]
> What are the performance implications?

It's by remark author so I would not worry too much about that.

> What are the advantages over what we currently have?

- Short version: it's high-fidelity/quality highlighting
- Long version: read its readme

> Given that this is a big breaking change, we should consider doing #55 at the same time

It's a breaking change for good. And this is intended after #55, or else it would need to import styles from starry-night.

**Comment by lino-levan** [comment]
We determined in #67 that starry night is too slow for default usage here. We *might* be able to still use it if we have a toggle between "fast but bad" vs "good but slow".

**Comment by kyeotic** [comment]
Since [YAML parsing is still broken](https://github.com/denoland/deno-gfm/issues/60), I would very much appreciate such a toggle

**Comment by lino-levan** [comment]
@hashrock opinions on a toggle here?

**Comment by kyeotic** [comment]
I don't want to say I don't care about performance _at all_, because if the page took 20 seconds to load it would be a dealbreaker. If it takes 2 seconds to load a page with _bullerproof_ syntax highlighting that matches GitHub _exactly_, though, I'm good with that.

I care way more about accuracy than performance for a blog. Looking at [all the YAML blocks with only two colors](https://blog.kye.dev/proxmox-servarr-stack#docker-compose), and most of them wrong, is aggravating.

I just spent 2 hours trying to move to Starry Night myself, but got so caught up trying to match the the styling that is spread between UnoCSS, gfm, and the syntax css that I gave up. I'll probably take another crack soon, because after seeing what Starry Night can I don't see any point to Prism anymore. Who cares if its _faster_ when it renders the _wrong thing_?

**Comment by lino-levan** [comment]
I've used Starry Night extensively in Pyro. The performance in real-world code snippets is very tolerable (sub 100 ms for a whole page). The performance on arbitrary code snippets (minified code or 1000+ lines of code) is rather slow. I think for personal blog posts or documentation where the code snippets are known a priori, starry night works great.

**Comment by kyeotic** [comment]
Yeah, that looks perfect. I'm not yet ready to move to a different platform, but if getting Starry Night working with deno-gfm/deno_blog ends up being too much work I will give Pyro a look.

**Comment by lino-levan** [comment]
Pyro was just an example use of Starry Night, it's really great for nice looking code snippets. I do not recommend Pyro for blogs just yet. I'm +1 for having a config option for starry night. What are your thoughts hashrock?

**Comment by hashrock** [comment]
Hmm I don't have strong opinion...
Is it difficult to allow users to choose between using 'starry-night' and 'prism'?

**Comment by hashrock** [comment]
yeah +1 for having a config option

**Comment by kyeotic** [comment]
If you do add a config option, please include `all` vs `common` for the Starry Night language selection.
